Fill.py

```diff
@@ -15,8 +17,71 @@ class FillError(RuntimeError):
     pass
 
 
-def _log_fill_progress(name: str, placed: int, total_items: int) -> None:
-    logging.info(f"Current fill step ({name}) at {placed}/{total_items} items placed.")
+class FillLogger():
+    min_size: int = 1000   # minimum number of items we care about reporting (non-tty)
+    min_time: float = 0.25 # minimum time (in seconds) between reports (tty)
+
+    def __init__(self, total_items: int):
+        self.start_time = time.time()
```

Comment:
not a big deal since this is technically correct, but general AP style is to do all the class attribute type annotations in the class definition, then assign to them in the constructor. it looks like `step` and `is_debug` could be assigned in the class body but i'd personally prefer
```py
self.is_debug = logging.getLogger().isEnabledFor(logging.DEBUG)
```

Reply:
I believe this is done. Please verify.